src/base/settingsstorage.cpp

```diff
@@ -182,26 +155,43 @@ QVariantHash TransactionalSettings::read() const
     }
     else
     {
-        deserialize(m_name, res);
+        deserialize(m_data, m_nativeSettingsName);
     }
-
-    return res;
 }
 
-bool TransactionalSettings::write(const QVariantHash &data) const
+bool SettingsStorage::writeNativeSettings() const
 {
+    std::unique_ptr<QSettings> nativeSettings = Profile::instance()->applicationSettings(m_nativeSettingsName + u"_new");
+
     // QSettings deletes the file before writing it out. This can result in problems
     // if the disk is full or a power outage occurs. Those events might occur
     // between deleting the file and recreating it. This is a safety measure.
     // Write everything to qBittorrent_new.ini/qBittorrent_new.conf and if it succeeds
     // replace qBittorrent.ini/qBittorrent.conf with it.
-    const Path newPath = serialize(m_name + u"_new", data);
-    if (newPath.isEmpty())
+    for (auto i = m_data.begin(); i != m_data.end(); ++i)
+        nativeSettings->setValue(i.key(), i.value());
+
+    nativeSettings->sync(); // Important to get error status
+
+    switch (nativeSettings->status())
     {
-        Utils::Fs::removeFile(newPath);
```

Comment:
Please preserve this action.

Reply:
> Please preserve this action.

Did you see the all previous code?
```c++
if (newPath.isEmpty())
{
    Utils::Fs::removeFile(newPath);
    return false;
}
```
It was meaningless to remove file with empty name so I drop this line.
I didn't change any logic. I just reformat the code. I would prefer not to touch the logic in this PR.

Reply:
Damn, it is supposed to remove the file when serialization goes wrong... I can handle it later if you preferred not to touch it.

Reply:
> I can handle it later if you preferred not to touch it.

👍